### PR TITLE
Handle lint errors that do not include location

### PR DESCRIPTION
**Pull request by piotrkubisa**
Helps to get rid of problems with go-vet errors, essentially those which points a problem related to resolving a package with missing go.mod file.

### CODE REVIEW

File: go/lint.py [hunk 1]
```diff
@@ -128,8 +128,20 @@ def parse(line, filename, tool):
         file = file[2:]
 
       if not parts[1].isdigit():
-        row = int(parts[0])
         msg = parts[1]
+
+        try:
+          row = int(parts[0])
+        except ValueError:
+          # In case of missing position, display an error in the first line
```
Comment (yields) [inline go/lint.py:136]: should we show the error next to the cursor, just thinking of larger files where you won't notice the first line.

**Comment by yields** [comment]
oh nice catch, this lgtm, just added a single note.

**Comment by yields** [comment]
oh also, is there a reliable way to reproduce this issue?

**Comment by piotrkubisa** [comment]
Personally, I am not sure if such fix is complete, because `go-vet` also works without being surrounded by  `go.mod` or `$GOPATH`. This problem is because we try to resolve a package (by absolute path) not a file.

**Comment by yields** [comment]
ah yeah, I think i caught this happening, basically the plugin didn't work at all unless I added a go.mod file.

maybe `buffer.package(view), buffer.file(view)` should work differently if `go.mod` does not exist?

**Comment by piotrkubisa** [comment]
> oh also, is there a reliable way to reproduce this issue?

1. `cd ~/Downloads` 
1. `git clone https://github.com/yields/sublime-go.git`
1. `cd sublime-go`
1. `git checkout a0f76fdc4c76a07148f06900e82034a52ce7572d` // not actually needed, last commit
1. `subl ./`
1. Open `examples/tags.go`.
1. Add some func with unused variable.
1. Save to trigger `go-vet`.

Console output:

```
Traceback (most recent call last):
  File "<snip>/Golang.sublime-package/go/decorators.py", line 51, in <lambda>
  File "<snip>/Golang.sublime-package/go/decorators.py", line 57, in call
  File "<snip>/Golang.sublime-package/go/decorators.py", line 12, in tracer
  File "<snip>/Golang.sublime-package/go/vet.py", line 33, in run
  File "<snip>/Golang.sublime-package/go/lint.py", line 41, in parse
  File "<snip>/Golang.sublime-package/go/lint.py", line 127, in parse
ValueError: invalid literal for int() with base 10: 'package /d/sublime-go/examples'
```

**Comment by yields** [comment]
This doesn't seem to happen on master, and it also includes some other fixes so I just released v0.0.6, not sure how to make it show up in package control (IIRC it should automatically show up?)

but until it shows up, you can install directly from master / v0.0.6 tag to see if this is still happening to you.

**Comment by piotrkubisa** [comment]
> ah yeah, I think i caught this happening, basically the plugin didn't work at all unless I added a go.mod file.
> 
> maybe `buffer.package(view), buffer.file(view)` should work differently if `go.mod` does not exist?

Yeah, I think we should take a look at it, it might be useful in future, however it's quite an edge-case for most of Go repositories. In my humble opinion it would be better to invest free time in development of more classy IDE-like features. Although, it should be noted this problem exist.

**Comment by piotrkubisa** [comment]
> This doesn't seem to happen on master, and it also includes some other fixes so I just released v0.0.6, not sure how to make it show up in package control (IIRC it should automatically show up?)
> 
> but until it shows up, you can install directly from master / v0.0.6 tag to see if this is still happening to you.

The issue is still actual, sadly :cry: `go-vet` reports a problem it cannot load a package.

File: go/lint.py (hunk 1 above)
Comment (piotrkubisa) [reply go/lint.py:136]: Good idea! You merely have a chance to notice that error even with 100-lines file.

**Comment by yields** [comment]
hmm interesting, I was sure it was fixed in master since i tested on v0.0.5 from package control and then on master.

I'll take a deeper look later on.

**Comment by piotrkubisa** [comment]
btw. won't be better to have `go-vet` and `golint` linting done using SublimeLinter? It's opinionated but thanks to it is straightforward to handle them.

// Yup, the curently `SublimeLinter-govet` does seem behind `sublime-go` :stuck_out_tongue:

**Comment by piotrkubisa** [comment]
> hmm interesting, I was sure it was fixed in master since i tested on v0.0.5 from package control and then on master.
> 
> I'll take a deeper look later on.

I suppose you have noticed it when you run `subl main.go` as you noted such situation might happen - see https://github.com/yields/sublime-go/blob/dc43d567b485fa8bd1643965697b97bdcfa950d3/go/vet.py#L25-L26

However if you open a folder that does not consist a `go.mod` a `go-vet` will report a problem with analysing the package.

Interestingly `vscode-go` (https://github.com/microsoft/vscode-go/blob/fc63f6a4f7126e9a239df44df8ee552bfefd23ff/src/goVet.ts#L73-L76) use `.` and `./...` wildcards, which seems to cover 99% situations. If we would like to change it appropriately we will need also change the `cwd` in govet's  `exec.Command("go", args=args, cwd=cwd)` and also `lint.py` which will properly recognize errors from linter and map to proper file (in multi-folder projects `vet` will report `./filename.go` file, while sublime will require `my-api/pkg/server/filename.go ).

**Comment by yields** [comment]
I see, thanks for looking into this man, I think if we want to go with relative as opposed to absolute paths we can simply adjust `buffer.package(view)` and `buffer.filename(view)` to return a relative path in all situations.

**Comment by piotrkubisa** [comment]
> I think if we want to go with relative as opposed to absolute paths we can simply adjust buffer.package(view) and buffer.filename(view) to return a relative path in all situations.

I think it is wrong way, relatives are error prone and many go linters even prefer absolute paths instead of relative.

File: go/lint.py (hunk 1 above)
Comment (piotrkubisa) [reply go/lint.py:136]: Well, actually the problem is from, there is no variable passed from where I might get the current position of the view :thinking:

**Comment by piotrkubisa** [comment]
I am closing this PR. If you think it is worth revisiting - feel free to reopen it.

**Pull request closed without merging**